docs/hyperopt.md

```diff
@@ -1,153 +1,109 @@
-cannot use your config file. It is also made on purpose to allow you
-testing your strategy with different configurations.
-
-The Hyperopt configuration is located in 
-[user_data/hyperopt_conf.py](https://github.com/freqtrade/freqtrade/blob/develop/user_data/hyperopt_conf.py).
-
-
-## Advanced notions
-### Understand the Guards and Triggers
-When you need to add the new guards and triggers to be hyperopt 
-parameters, you do this by adding them into the [hyperopt_space()](https://github.com/freqtrade/freqtrade/blob/develop/user_data/strategies/test_strategy.py#L244-L297).
-
-If it's a trigger, you add one line to the 'trigger' choice group and that's it.
-
-If it's a guard, you will add a line like this:
-```
-'rsi': hp.choice('rsi', [
-    {'enabled': False},
-    {'enabled': True, 'value': hp.quniform('rsi-value', 20, 40, 1)}
-]),
+    def indicator_space() -> List[Dimension]:
+        """
+        Define your Hyperopt space for searching strategy parameters
+        """
+        return [
+            Integer(20, 40, name='adx-value'),
+            Integer(20, 40, name='rsi-value'),
+            Categorical([True, False], name='adx-enabled'),
+            Categorical([True, False], name='rsi-enabled'),
+            Categorical(['bb_lower', 'macd_cross_signal'], name='trigger')
+        ]
 ```
-This says, "*one of the guards is RSI, it can have two values, enabled or 
-disabled. If it is enabled, try different values for it between 20 and 40*".
 
-So, the part of the strategy builder using the above setting looks like 
-this:
+Above definition says: I have five parameters I want you to randomly combine 
+to find the best combination. Two of them are integer values (`adx-value` 
+and `rsi-value`) and I want you test in the range of values 20 to 40.
+Then we have three category variables. First two are either `True` or `False`.
+We use these to either enable or disable the ADX and RSI guards. The last
+one we call `trigger` and use it to decide which buy trigger we want to use.
+
+So let's write the buy strategy using these values:
 
 ```
-if params['rsi']['enabled']:
-    conditions.append(dataframe['rsi'] < params['rsi']['value'])
+        def populate_buy_trend(dataframe: DataFrame) -> DataFrame:
+            conditions = []
+            # GUARDS AND TRENDS
+            if 'adx-enabled' in params and params['adx-enabled']:
+                conditions.append(dataframe['adx'] > params['adx-value'])
+            if 'rsi-enabled' in params and params['rsi-enabled']:
+                conditions.append(dataframe['rsi'] < params['rsi-value'])
+
+            # TRIGGERS
+            if params['trigger'] == 'bb_lower':
+                conditions.append(dataframe['close'] < dataframe['bb_lowerband'])
+            if params['trigger'] == 'macd_cross_signal':
+                conditions.append(qtpylib.crossed_above(
+                    dataframe['macd'], dataframe['macdsignal']
+                ))
+
+            dataframe.loc[
```

Comment:
can we add a comment above this that the below line here and in `hyperopt.py` that this is a "do not change" line?